### PR TITLE
feat: start datanode with config

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Be able to start the new datanode with config file.

I add a new common subcrate `common-config` for all config structs. This PR only moves `WalConfig` here, there will be a follow up PR to move all other structs

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: config/datanode.example.toml [hunk 1]
```diff
@@ -71,10 +69,21 @@ auto_flush_interval = "1h"
...
+[[store_engine]]
+[store_engine.mito]
+# Number of region workers
+num_workers = 8
+# Request channel size of each worker
+worker_channel_size = 128
+# Max batch size for a worker to handle requests
+worker_request_batch_size = 64
+# Number of meta action updated to trigger a new checkpoint for the manifest
+manifest_checkpoint_distance = 10
```
Comment (v0y4g3r) [inline config/datanode.example.toml:82]: I'd prefer `manifest_checkpoint_interval` over `manifest_checkpoint_distance`

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: config/datanode.example.toml (hunk 1 above)
Comment (waynexia) [reply config/datanode.example.toml:82]: 🥵 this name has been argued a lot...

File: src/common/config/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+use std::time::Duration;
+
+use common_base::readable_size::ReadableSize;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(default)]
+pub struct WalConfig {
+    // wal directory
+    pub dir: Option<String>,
```
Comment (evenyag) [inline src/common/config/src/lib.rs:24]:
```suggestion
```

File: config/datanode.example.toml [hunk 3]
```diff
@@ -71,10 +69,21 @@ auto_flush_interval = "1h"
 # Global write buffer size for all regions.
 global_write_buffer_size = "1GB"
 
-# Procedure storage options, see `standalone.example.toml`.
-[procedure]
-max_retry_times = 3
-retry_delay = "500ms"
+# Mito engine options
+[[store_engine]]
+[store_engine.mito]
```
Comment (evenyag) [inline config/datanode.example.toml:74]:
```suggestion
[[region_engine]]
[region_engine.mito]
```

**Review by evenyag — APPROVED** [review]

**Pull request merged**